multicluster/cmd/multicluster-controller/clusterset_webhook.go

```diff
@@ -57,6 +63,22 @@ func (v *clusterSetValidator) Handle(ctx context.Context, req admission.Request)
 			klog.ErrorS(err, "the field 'clusterID' is immutable", "ClusterSet", klog.KObj(clusterSet))
 			return admission.Denied("the field 'clusterID' is immutable")
 		}
+
+		if isMemberListChanged(oldClusterSet.Spec.Members, clusterSet.Spec.Members) {
```

Comment:
Do you think more efficient to check ServiceAccount name matches or not, before checking member list changes? Not sure.

Reply:
I think other users may still update other parts of ClusterSet, eg: add a label or an annotation etc.

Reply:
I meant if checking ServiceAccount name is faster than checking member list, maybe we should check SA name first, also assuming more updates should be by MC controller.

Reply:
You mean we allow user to create/delete ClusterSet, but for update, we assume it will only be done by MC controller?

Reply:
No, I meant if the current code is more efficient or not:

```
		ui := req.UserInfo
		_, saName, err := serviceaccount.SplitUsername(ui.Username)
		if err != nil {
			klog.ErrorS(err, "Error getting ServiceAccount name", "ClusterSet", req.Namespace+"/"+req.Name)
			return admission.Errored(http.StatusBadRequest, err)
		}
		if saName != mcControllerSAName && isMemberListChanged(oldClusterSet.Spec.Members, clusterSet.Spec.Members) {
			return admission.Errored(http.StatusPreconditionFailed, fmt.Errorf("member list can only be updated by Antrea Multi-cluster Controller"))
		}
```

assuming most updates are by controller.

Reply:
Refined to check service account first.